CorsixTH/Lua/dialogs/resizables/cheats_dialog.lua

```diff
@@ -91,20 +97,37 @@ function UICheats:UICheats(ui)
   self.item_buttons = {}
 
   y = y + 30
-  for num = 1, #self.cheat_list do
-    self.item_panels[num] = self:addBevelPanel(20, y, 260, 20, col_bg)
-      :setLabel(_S.cheats_window.cheats[self.cheat_list[num].name])
-    self.item_buttons[num] = self.item_panels[num]:makeButton(0, 0, 260, 20, nil, button_clicked(num))
-      :setTooltip(_S.tooltip.cheats_window.cheats[self.cheat_list[num].name])
-    y = y + 20
+  local num = 1
+  while self.cheat_list[num] do
+    if second_col then
+      self.item_panels[num] = self:addBevelPanel(20, y, 260, 20, col_bg)
+        :setLabel(_S.cheats_window.cheats[self.cheat_list[num].name])
+      self.item_buttons[num] = self.item_panels[num]:makeButton(0, 0, 260, 20, nil, button_clicked(num))
+        :setTooltip(_S.tooltip.cheats_window.cheats[self.cheat_list[num].name])
+      num = num + 1
+      if self.cheat_list[num] then
+        self.item_panels[num] = self:addBevelPanel(second_col, y, 260, 20, col_bg)
+          :setLabel(_S.cheats_window.cheats[self.cheat_list[num].name])
+        self.item_buttons[num] = self.item_panels[num]:makeButton(0, 0, 260, 20, nil, button_clicked(num))
+          :setTooltip(_S.tooltip.cheats_window.cheats[self.cheat_list[num].name])
+      end
+      y = y + 20
+    else
+      self.item_panels[num] = self:addBevelPanel(20, y, 260, 20, col_bg)
+        :setLabel(_S.cheats_window.cheats[self.cheat_list[num].name])
+      self.item_buttons[num] = self.item_panels[num]:makeButton(0, 0, 260, 20, nil, button_clicked(num))
+        :setTooltip(_S.tooltip.cheats_window.cheats[self.cheat_list[num].name])
```

Comment:
You add this code 3 times, stick it in a local function and call that 3 times.

In that way, it's much simpler to understand what is unique in these calls and what is always the same. Also, if some value needs to be changed, you can do that with 1 change rather than 3 changes with all the risks that it becomes inconsistent.